client/src/pages/messaging/index.js

```diff
@@ -139,17 +139,25 @@ const Chat = () => {
   );
 
   const getSetUsers = async (channelID) => {
-    const usersInChannel = await getUsersInChannel({ channel: channelID });
+    try {
+      const usersInChannel = await getUsersInChannel({ channel: channelID });
 
-    setUsers(usersInChannel);
-    const alice = usersInChannel.find((user) => user.uuid !== userId);
+      setUsers(usersInChannel);
+      const alice = usersInChannel.find((user) => user.uuid !== userId);
 
-    // if alice is already connected,
-    // get alice's publicKey
-    if (alice) {
-      const key = await getPublicKey({ userId: alice.uuid, channel: channelID });
-      publicKeyRef.current = strToTypedArr(key.publicKey);
-      playNotification();
+      // if alice is already connected,
+      // get alice's publicKey
+      if (alice) {
+        const key = await getPublicKey({ userId: alice.uuid, channel: channelID });
+        publicKeyRef.current = strToTypedArr(key.publicKey);
+        playNotification();
+      }
+    } catch (err) {
+      if (err.status === 404) {
+        setUsers([]);
+      } else {
+        console.error(err);
+      }
```

Comment:
My refactoring suggestion would be: 
```
let usersInChannel = [];
try {
    usersInChannel = await getUsersInChannel({ channel: channelID });
} catch (err ) {
    // log some error
}
...
rest is the same as before
```

Reply:
if the getUsersInChannel returns error and the usersInChannel is an empty array then the find operation on usersInChannel to find Alice and the code after to handle Alice key isn't relevant, so this is why I put all the code inside the try block.

Reply:
Right, but it's to keep the code flat, more readable (top-bottom). And, the find ops is not costly in this case because the array is empty. Also, it doesn't go below the line `if (alice)`

Reply:
I refactored the try-catch as you suggested but with const and spread operator.
I have learned to avoid as much as I can from let in JS is BP, let me know what do you think about this approach.

Reply:
You're right, we should avoid `let` as much as possible so that we don't **accidentally** mutate a variable and introduce an unwanted bug.  But here, `let` is a clear indication that the variable will be reassigned (mutated). I don't see any reason to avoid using `let` here. 

Your approach is totally fine here, however, let's keep the assignment in a separate line for the same reason - flat, and readability. 
Why do you need spread `...` here?   
```
   try {
          const users = await getUsersInChannel({ channel: channelID });
          usersInChannel.push(users);
    } catch (err) {
          console.error(err);
    }
```

Reply:
> Why do you need spread `...` here?

because the getUsersInChannel returns array, If we push array to the array we get array of arrays, I use spread to add the elements of the users into usersInChannel